pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,145 @@ type ExternalNodeList struct {
+	// BundlePartiallySucceeded means some required Nodes or ExternalNodes have successfully
+	// processed the SupportBundleCollection request, but others failed.
+	BundlePartiallySucceeded BundlePhase = "PartiallySucceeded"
+	// BundleFailed means the SupportBundleCollection failed. It can happen when Antrea
+	// Controller fails to create internal objects, or all of the required Nodes or
+	// ExternalNodes fail to process the request.
+	BundleFailed BundlePhase = "Failed"
+)
+
+type SupportBundleCollectionStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// The number of Nodes and ExternalNodes that have completed the SupportBundleCollection.
+	SucceededNodes int32 `json:"succeededNodes"`
+	// The total number of Nodes and ExternalNodes that should process the SupportBundleCollection.
+	DesiredNodes int32 `json:"desiredNodes"`
+	// Represents the latest available observations of a SupportBundleCollection current state.
+	Conditions []SupportBundleCollectionCondition `json:"conditions"`
+}
+
+type SupportBundleCollectionConditionType string
+
+const (
+	// Started means Antrea Controller has created the internal resource for the SupportBundleCollectionCondition.
+	Started SupportBundleCollectionConditionType = "started"
+	// Collected means all Antrea Agents have successfully processed the SupportBundleCollectionCondition request.
+	Collected SupportBundleCollectionConditionType = "collected"
+)
+
+// SupportBundleCollectionCondition describes the state of a SupportBundleCollection at a certain point.
+type SupportBundleCollectionCondition struct {
+	// Type of StatefulSet condition.
+	Type SupportBundleCollectionConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status metav1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// A human-readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNodes struct {
+	Namespace string
+	// List the names of certain ExternalNodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain ExternalNodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
```

Comment:
How to collect support bundle of all nodes in all namespaces?

Reply:
Do you mean collect support bundle of all ExternalNodes in all namespaces? User should define different CRD request to collect one namespace by another. I don't think it is a valid request since ExternalNode is working in Namespace scope. From troubleshooting perspective, the bundle files should also be in namespace scope.